#!/bin/sh -e
## 129_dir_section.dpatch
##
## DP: Description: Add directory section for info documents.
## DP: Author: Matthias Klose
## DP: Upstream status: submitted upstream

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

bfd/
2009-08-27  Matthias Klose  <doko@ubuntu.com>

	* bfd.texinfo: Add directory section for info document.

gas/
2009-08-27  Matthias Klose  <doko@ubuntu.com>

	* doc/as.texinfo: Add directory section for info document.

gprof/
2009-08-27  Matthias Klose  <doko@ubuntu.com>

	* gprof.texi: Add directory section for info document.

ld/
2009-08-27  Matthias Klose  <doko@ubuntu.com>

	* ld.texinfo, ldint.texinfo: Add directory section for info document.

@DPATCH@
--- ./bfd/doc/bfd.texinfo~	2008-11-19 17:22:46.000000000 +0100
+++ ./bfd/doc/bfd.texinfo	2009-08-27 15:59:58.000000000 +0200
@@ -6,13 +6,12 @@
 @c 
 @synindex fn cp
 
-@ifinfo
-@format
-START-INFO-DIR-ENTRY
+@ifnottex
+@dircategory Software development
+@direntry
 * Bfd: (bfd).                   The Binary File Descriptor library.
-END-INFO-DIR-ENTRY
-@end format
-@end ifinfo
+@end direntry
+@end ifnottex
 
 @copying
 This file documents the BFD library.
--- ./gas/doc/as.texinfo~	2009-08-06 19:38:02.000000000 +0200
+++ ./gas/doc/as.texinfo	2009-08-27 15:59:27.000000000 +0200
@@ -84,14 +84,13 @@
 @c might as well show 'em anyways.
 @end ifinfo
 
-@ifinfo
-@format
-START-INFO-DIR-ENTRY
+@ifnottex
+@dircategory Software development
+@direntry
 * As: (as).                     The GNU assembler.
 * Gas: (as).                    The GNU assembler.
-END-INFO-DIR-ENTRY
-@end format
-@end ifinfo
+@end direntry
+@end ifnottex
 
 @finalout
 @syncodeindex ky cp
--- ./gprof/gprof.texi~	2009-08-27 15:26:11.000000000 +0200
+++ ./gprof/gprof.texi	2009-08-27 16:00:58.000000000 +0200
@@ -10,15 +10,14 @@
 @include bfdver.texi
 @c man end
 
-@ifinfo
+@ifnottex
 @c This is a dir.info fragment to support semi-automated addition of
 @c manuals to an info tree.  zoo@cygnus.com is developing this facility.
-@format
-START-INFO-DIR-ENTRY
+@dircategory Software development
+@direntry
 * gprof: (gprof).                Profiling your program's execution
-END-INFO-DIR-ENTRY
-@end format
-@end ifinfo
+@end direntry
+@end ifnottex
 
 @copying
 This file documents the gprof profiler of the GNU system.
--- ./ld/ld.texinfo~	2009-07-06 15:48:51.000000000 +0200
+++ ./ld/ld.texinfo	2009-08-27 15:57:09.000000000 +0200
@@ -39,13 +39,12 @@
 @end ifset
 @c man end
 
-@ifinfo
-@format
-START-INFO-DIR-ENTRY
+@ifnottex
+@dircategory Software development
+@direntry
 * Ld: (ld).                       The GNU linker.
-END-INFO-DIR-ENTRY
-@end format
-@end ifinfo
+@end direntry
+@end ifnottex
 
 @copying
 This file documents the @sc{gnu} linker LD
--- ./ld/ldint.texinfo~	2007-10-01 11:54:57.000000000 +0200
+++ ./ld/ldint.texinfo	2009-08-27 15:58:50.000000000 +0200
@@ -4,13 +4,12 @@
 @c 2003, 2007
 @c Free Software Foundation, Inc.
 
-@ifinfo
-@format
-START-INFO-DIR-ENTRY
+@ifnottex
+@dircategory Software development
+@direntry
 * Ld-Internals: (ldint).	The GNU linker internals.
-END-INFO-DIR-ENTRY
-@end format
-@end ifinfo
+@end direntry
+@end ifnottex
 
 @copying
 This file documents the internals of the GNU linker ld.
